src/marqo/s2_inference/clip_utils.py

```diff
@@ -139,10 +139,15 @@ def download_image_from_url(image_path: str, image_download_headers: dict, timeo
     if not isinstance(timeout_ms, int):
         raise InternalError(f"timeout must be an integer but received {timeout_ms} of type {type(timeout_ms)}")
 
+    try:
+        encoded_url = requote_uri(image_path)
```

Comment:
Is it possible that image_path is already encoded? How do we tell?

Reply:
This function passes the given URL through an unquote/quote cycle to ensure that it is fully and consistently quoted. There are unit tests on these cases (URL with encoded characters, URL with mixed encoded and unencoded characters)